integration/xgboost/encryption_plugins/src/include/base_plugin.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>     // for uint8_t, uint32_t, int32_t, int64_t
+#include <string_view> // for string_view
+#include <utility>     // for pair
+#include <vector>      // for vector
+#include <sstream>
+#include <iomanip>
+
+#include "util.h"
+
+namespace nvflare {
+
+// Plugin that uses Python tenseal and GRPC.
+class BasePlugin {
```

Comment:
docstring needs to be updated, this could be anything, could also just use cuda or C++ code.

Reply:
added docstring to the class.